src/processes/helper/addUserToTopicChannel.js

```diff
@@ -1,41 +1,39 @@
-const { convertString, ICON_TOPIC_CHANNEL, CHANNEL_TYPE_TOPIC } = require('../../utils');
-const { LogError } = require('../../databases/models');
-const StreamChat = require("stream-chat").StreamChat;
+const {convertString, ICON_TOPIC_CHANNEL, CHANNEL_TYPE_TOPIC} = require('../../utils');
+const {LogError} = require('../../databases/models');
+const StreamChat = require('stream-chat').StreamChat;
 
 const addUserToTopicChannel = async (user_id, topics) => {
-    try {
-        let userId = user_id.toString();
-        const serverClient = StreamChat.getInstance(
-            process.env.API_KEY,
-            process.env.SECRET
-        );
-        topics.map(async (item) => {
-            serverClient.createToken(userId);
-            const channelId = item;
-            const members = [];
-            members.push(userId);
+  try {
+    let userId = user_id.toString();
+    const serverClient = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+    topics.map(async (topic) => {
+      const newTopic = convertString(topic, '-', '').toLowerCase();
+      const token = serverClient.createToken(userId);
 
-            let channelName = "#" + convertString(item, "-", "");
+      await serverClient.connectUser({id: userId}, token);
+      const members = [];
+      members.push(userId);
 
-            const channel = serverClient.channel("topics", channelId, {
-                name: channelName,
-                created_by_id: "system",
-                channel_type: CHANNEL_TYPE_TOPIC,
-                channelImage: ICON_TOPIC_CHANNEL,
-                channel_image: ICON_TOPIC_CHANNEL,
-                image: ICON_TOPIC_CHANNEL
-            });
-            await channel.create();
+      const channel = serverClient.channel('topics', `topic_${newTopic}`, {
+        name: `#${newTopic}`,
+        channel_type: CHANNEL_TYPE_TOPIC,
+        channelImage: ICON_TOPIC_CHANNEL,
+        channel_image: ICON_TOPIC_CHANNEL,
+        image: ICON_TOPIC_CHANNEL
+      });
+      await channel.create();
 
-            await channel.addMembers(members);
-
-        });
-    } catch (error) {
-        await LogError.create({
-            message: error.message
-        });
-        console.error('addUserToTopicChannel: ', error);
-    }
+      await channel.addMembers(members);
+      await channel.sendMessage({text: 'This topic has new followers'}, {skip_push: true});
+      console.log('channel followed');
+      serverClient.disconnect();
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `map` function is used here, which returns a new array. However, the intention seems to be to perform side effects (i.e., creating channels and adding members) rather than generating a new array. It would be more appropriate to use `forEach` instead of `map`. Also, since you're using async operations inside the loop, it's better to use `for...of` with `Promise.all` to ensure all async operations are completed before proceeding.

```diff
-    topics.map(async (topic) => {
+    const promises = topics.map(async (topic) => {
```

And at the end of the block:

```diff
-      serverClient.disconnect();
+      return serverClient.disconnect();
    });
+    await Promise.all(promises);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->